File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <vector>

using namespace std;

int stack[10001];
int index = -1;

void push(int X);
int pop(void);
int size(void);
int empty(void);
int top(void);

int main(void) {
    string command;
    int num, input;
    
    cin >> input;
    while (input--) {
        cin >> command;
        if (command == "push") {
            cin >> num;
            push(num);
        } else if (command == "pop") {
            cout << pop() << endl;
        } else if (command == "size") {
            cout << size() << endl;
        } else if (command == "empty") {
            cout << empty() << endl;
        } else if (command == "top") {
            cout << top() << endl;
        }
    }
    return 0;
}

void push(int X) {
    stack[++index] = X;
} 

int pop(void) {
    if (index < 0) {
        return -1;
    }
    return stack[index--];
}

int size(void) {
    if (index < 0) {
        return 0;
    }
    return index + 1;
}

int empty(void) {
    if (index < 0) {
        return 1;
    }
    return 0;
}

int top(void) {
    if (index < 0) {
        return -1;
    }
    return stack[index];
}

File: 백준/Silver/10828. 스택/스택.java
import java.util.Scanner;
class Stack {
    private int stack[] = new int[10001];
    private int top = 0;
    
    public void push(int x) {
        stack[top++] = x;
    }
    
    public int pop() {
        if (top == 0) {
            return -1;
        }
        return stack[--top];
    }
    
    public int size() {
        return top;
    }
    
    public int empty() {
        if (top == 0) {
            return 1;
        }
        return 0;
    }
    
    public int top() {
        if (top == 0) {
            return -1;
        }
        return stack[top - 1];
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Stack stack = new Stack();
        
        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            String command = sc.next();
            switch(command) {
                case "push" :
                    int x = sc.nextInt();
                    stack.push(x);
                    break;
                case "pop" :
                    System.out.println(stack.pop());
                    break;
                case "size" :
                    System.out.println(stack.size());
                    break;
                case "empty" :
                    System.out.println(stack.empty());
                    break;
                case "top" :
                    System.out.println(stack.top());
                    break;
            }
        }
    }
}